### PR TITLE
Update project for Xcode 15

### PR DESCRIPTION
**Pull request by pepicrft**
### Short description 📝
GitHub's [hosted runner](https://github.com/actions/runner-images/blob/main/images/macos/macos-13-Readme.md#xcode) has Xcode 15 already. This PR prepares the project to use Xcode 15 and updates the CI pipelines to use that version.

### How to test the changes locally 🧐
Common workflows such as a build and testing should work.

### Contributor checklist ✅

- [x] The code has been linted using run `./fourier lint tuist --fix`
- [ ] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [ ] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

**Review by danyf90 — COMMENTED** [review]
I think you need to change all the GitHub action configs to use the new runners?

**Comment by pepicrft** [comment]
> I think you need to change all the GitHub action configs to use the new runners?

If I'm not wrong, I think it's already pre-installed in the default runners

File: Sources/TuistGenerator/Utils/EmbedScriptGenerator.swift [hunk 1]
```diff
@@ -130,7 +130,7 @@ final class EmbedScriptGenerator: EmbedScriptGenerating {
         echo "mkdir -p ${CONFIGURATION_BUILD_DIR}/${FRAMEWORKS_FOLDER_PATH}"
         mkdir -p "${CONFIGURATION_BUILD_DIR}/${FRAMEWORKS_FOLDER_PATH}"
 
-        SWIFT_STDLIB_PATH="${DT_TOOLCHAIN_DIR}/usr/lib/swift/${PLATFORM_NAME}"
+        SWIFT_STDLIB_PATH="${TOOLCHAIN_DIR}/usr/lib/swift/${PLATFORM_NAME}"
```
Comment (pepicrft) [inline Sources/TuistGenerator/Utils/EmbedScriptGenerator.swift:133]: For context, [`DT_TOOLCHAIN_DIR`](https://github.com/CocoaPods/CocoaPods/pull/12009) causes issues with Xcode 15.

**Review by fortmarek — APPROVED** [review]
Tested this on my local machine with Xcode 15.0. Once we get the CI green, LGTM 🚀 👏

**Pull request merged**